### PR TITLE
Replace controller-runtime.Manager custom mock with mockery generated Manager

### PR DESCRIPTION
**Pull request by aorcholski**
## Description
Replaces `controller-runtime.Manager` and `cmd.manager.Provider` custom mocks with mockery generated structs.

## How can this be tested?
Run unit tests.

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: cmd/csi/provisioner/builder_test.go [hunk 1]
```diff
@@ -4,9 +4,9 @@ import (
 	"io/fs"
 	"testing"
 
-	cmdManager "github.com/Dynatrace/dynatrace-operator/cmd/manager"
 	dtcsi "github.com/Dynatrace/dynatrace-operator/pkg/controllers/csi"
-	"github.com/Dynatrace/dynatrace-operator/test/mocks/cmd/config"
+	mocks "github.com/Dynatrace/dynatrace-operator/test/mocks/cmd/config"
+	mock "github.com/Dynatrace/dynatrace-operator/test/mocks/cmd/manager"
```
Comment (gkrenn) [inline cmd/csi/provisioner/builder_test.go:9]: Would use more meaningful names here, if package names clash

**Review by gkrenn — DISMISSED** [review]
See comments

File: cmd/csi/provisioner/command_test.go [hunk 2]
```diff
@@ -3,11 +3,12 @@ package provisioner
...
 	"testing"
 
-	cmdManager "github.com/Dynatrace/dynatrace-operator/cmd/manager"
 	dtfake "github.com/Dynatrace/dynatrace-operator/pkg/api/scheme/fake"
 	dtcsi "github.com/Dynatrace/dynatrace-operator/pkg/controllers/csi"
 	"github.com/Dynatrace/dynatrace-operator/pkg/util/logger"
-	"github.com/Dynatrace/dynatrace-operator/test/mocks/cmd/config"
+	mocks "github.com/Dynatrace/dynatrace-operator/test/mocks/cmd/config"
+	mockedprovider "github.com/Dynatrace/dynatrace-operator/test/mocks/cmd/manager"
+	mockedmanager "github.com/Dynatrace/dynatrace-operator/test/mocks/sigs.k8s.io/controller-runtime/pkg/manager"
```
Comment (albertogdd) [inline cmd/csi/provisioner/command_test.go:11]: IMO as package alias would be better to call them `providermock` and `managermock`

File: cmd/csi/provisioner/builder_test.go (hunk 1 above)
Comment (albertogdd) [reply cmd/csi/provisioner/builder_test.go:9]: +1

File: cmd/csi/provisioner/command_test.go (hunk 2 above)
Comment (albertogdd) [reply cmd/csi/provisioner/command_test.go:11]: And also change this one
`mocks "github.com/Dynatrace/dynatrace-operator/test/mocks/cmd/config"`
to 
`configmock "github.com/Dynatrace/dynatrace-operator/test/mocks/cmd/config"`

**Review by andriisoldatenko — APPROVED** [review]

**Review by albertogdd — APPROVED** [review]
Awesome :+1:

**Pull request merged**